src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnector.java

```diff
@@ -0,0 +1,170 @@
+import io.cdap.cdap.etl.api.connector.ConnectorSpec;
+import io.cdap.cdap.etl.api.connector.ConnectorSpecRequest;
+import io.cdap.cdap.etl.api.connector.PluginSpec;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import io.cdap.plugin.ariba.source.AribaBatchSource;
+import io.cdap.plugin.ariba.source.AribaServices;
+import io.cdap.plugin.ariba.source.config.AribaPluginConfig;
+import io.cdap.plugin.ariba.source.exception.AribaException;
+import io.cdap.plugin.ariba.source.metadata.AribaResponseContainer;
+import io.cdap.plugin.ariba.source.util.ResourceConstants;
+import io.cdap.plugin.common.ConfigUtil;
+import okhttp3.HttpUrl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+/**
+ * AribaConnector Class
+ */
+@Plugin(type = Connector.PLUGIN_TYPE)
+@Name(ResourceConstants.PLUGIN_NAME)
+@Description("Connection to access data from Ariba.")
+public class AribaConnector implements Connector {
+  private static final Logger LOG = LoggerFactory.getLogger(AribaConnector.class);
+  private static final String METADATA_PATH = "api/analytics-reporting-view/v1";
+  private static final String ENTITY_TYPE_TEMPLATE = "template";
+  private static final String VIEW_TEMPLATE = "viewTemplateName";
+  private static final String PATH_SEGMENT = "%s/viewTemplates";
+  private static final String RECORDS = "Records";
+  private static final String TOKEN = "PageToken";
+  private static final Gson gson = new Gson();
+  private final AribaConnectorConfig config;
+  private String accessToken;
+
+  public AribaConnector(AribaConnectorConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void test(ConnectorContext connectorContext) throws ValidationException {
+    FailureCollector collector = connectorContext.getFailureCollector();
+    config.validateCredentials(collector);
+    try {
+      config.validateToken(collector);
+    } catch (AribaException | IOException e) {
+      LOG.error(" Unable to validate token", e);
+    }
+  }
+
+  @Override
+  public BrowseDetail browse(ConnectorContext connectorContext, BrowseRequest browseRequest) throws IOException {
+    BrowseDetail.Builder browseDetailBuilder = BrowseDetail.builder();
+    int count = 0;
+    JsonArray jsonElements = new JsonArray();
+    JsonArray jsonArray = listTemplates(null, jsonElements);
+    for (JsonElement jsonElement : jsonArray) {
+      JsonObject resultJsonObject = gson.fromJson(jsonElement, JsonObject.class);
+      String name = resultJsonObject.get(VIEW_TEMPLATE).getAsString();
+      BrowseEntity.Builder entity = (BrowseEntity.builder(name, name, ENTITY_TYPE_TEMPLATE).
+        canBrowse(false).canSample(true));
```

Comment:
canSample(false) if we don't support sampling

Reply:
Making it false will freeze out the list, as mentioned in salesforce.

Reply:
Salesforce connector supports sampling, which is why it has canSample(true) - https://github.com/data-integrations/salesforce/blob/develop/src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java#L143

> Making it false will freeze out the list

What does this mean? Can you provide more details?

Reply:
It will display the list of templates but user will not be able to select anything from there, it's just to display the names.
Do you still want me to make it false with the above output?

Reply:
Is this expected behavior @yaojiefeng ?

Reply:
Chatted with Yaojie offline. This is a UI limitation, filed https://cdap.atlassian.net/browse/CDAP-20021 to track this issue.
Setting `canSample(true)` will likely break wrangler.
@Shubhangi-cs can we implement sampling for this connector? I understand that sampling will take a long time to return, but I think it's better than a broken experience for the user